credsweeper/file_handler/data_content_provider.py

```diff
@@ -32,6 +43,41 @@ def data(self, data: bytes) -> None:
         """data setter"""
         self.__data = data
 
+    def is_xml(self) -> bool:
+        """Tries to read data as xml
+        Return True if reading was successful
+        """
+        try:
+            xml_text = self.data.decode(encoding=DEFAULT_ENCODING).splitlines()
+            tree = etree.fromstringlist(xml_text)
+            for element in tree.iter():
+                tag = Util.extract_element_data(element, "tag")
+                text = Util.extract_element_data(element, "text")
+                self.lines.append(f"{tag} : {text}")
+        except Exception as exc:
+            logger.debug("Cannot parse as XML:%s %s", exc, self.data)
+            return False
```

Comment:
It seems duplicated code.. So I think this code block can be replace with below.

```suggestion
        lines, line_nums = Util.get_xml_data(file_path)
        if lines is None and line_nums is None:
            return False
        return True
```

And according to this change `Util.extract_element_data()` method should be change as before `Util._extract_element_data()` too.

Reply:
@csh519 
Review changes please

Reply:
I refactored StringContentProvider to apply parsed line numbers for raw data.